crc32.c

```diff
@@ -648,15 +649,7 @@ unsigned long ZEXPORT crc32_z(crc, buf, len)
     z_size_t last, last2, i;
     z_size_t num;
 
-    /* Return initial CRC, if requested. */
```

Comment:
I hooked into the portable code to save some code duplication.

As a result, for ARMv8 we will build both implementations (e.g. portable + optimized) and decide which one to use at runtime.

We also cache the CPU features detection flags to avoid wasting CPU cycles.